tests/test_routes.py

```diff
@@ -522,6 +534,112 @@ def test_delete_address_invalid_request(self):
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # update the Customer
+        dummy_email =  "User759@gmail.com";
+        updated_customer = response.get_json()
+        logging.debug(updated_customer)
+        updated_customer["email"] = dummy_email
+        response = self.client.put(f"{BASE_URL}/{updated_customer['id']}", json=updated_customer)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        updated_customer = response.get_json()
+        self.assertEqual(updated_customer["email"], dummy_email)
+
+    def test_update_invalid_customer(self):
+        """It should not Update Non existing Customer"""
+        customer = CustomerFactory()
+        customer.id = 10
+        customer_ser = customer.serialize()
+        logging.debug(customer)
+        response = self.client.put(f"{BASE_URL}/{customer_ser['id']}", json=customer_ser)
+        
+        self.assertEqual(response.status_code,status.HTTP_404_NOT_FOUND)
+
+    def test_update_customer_invalid_content_type(self):
+        """It should not Update an existing Customer with invalid content type request"""
+        # create a Customer to update
+        test_customer = CustomerFactory()
+        response = self.client.post(BASE_URL, json=test_customer.serialize())
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # update the Customer
+        dummy_email =  "User759@gmail.com";
+        updated_customer = response.get_json()
+        logging.debug(updated_customer)
+        updated_customer["email"] = dummy_email
+        response = self.client.put(f"{BASE_URL}/{updated_customer['id']}", json=updated_customer, content_type="custom_content_type_v100")
+        self.assertEqual(response.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+
+    ##########################################
+    # UPDATE ADDRESS
+    ##########################################
+    
+    def test_update_customer_address(self):
+        """It should Update an existing Customer Address"""
+        
+        # create a Customer with 3 addresses to update
+        customer = CustomerFactory()
+        addresses = AddressFactory.create_batch(3)
+        customer.addresses.extend(addresses)
+        customer.create()
+        customer_id = customer.id
+        logging.debug(customer)
+        # Check that customer been created
+        customer = Customer.find(customer_id)
+        self.assertIsNotNone(customer)
+        # check that addresses been created
+        self.assertEqual(len(customer.addresses), 3)
+        
+        # Pick an address randomly
+        address_id_to_update = random.choice(customer.addresses).address_id
+        # update the address
+        address = Address.find(address_id_to_update)
+        self.assertIsNotNone(address)
+        dummy_street =  "Washington Square";
+        logging.debug(address)
+        address = address.serialize()
+        address["street"] = dummy_street
```

Comment:
Same for addresses, if we can cover all fields updates.

Reply:
Marking as resolved, considering changes have been made